modules/workflow-service-impl/src/main/java/org/opencastproject/workflow/impl/WorkflowServiceImpl.java

```diff
@@ -2296,39 +2271,24 @@ private void removeWorkflowInstanceFromIndex(long workflowInstanceId, Elasticsea
   /**
    * Update a workflow instance in the API index.
    *
-   * @param workflowInstance
-   *         the workflowInstance to update
-   * @param accessControlList
-   *         the ACL of the event
-   * @param episodeDublincoreCatalog
-   *         the episode dublincore catalog of the event
+   * @param latestWorkflow
+   *         the workflowIndexData to update
    * @param index
    *         the index to update
    */
-  private void updateWorkflowInstanceInIndex(WorkflowInstance workflowInstance, AccessControlList accessControlList,
-          DublinCoreCatalog episodeDublincoreCatalog, ElasticsearchIndex index) {
-    final long workflowInstanceId = workflowInstance.getId();
-    final String eventId = workflowInstance.getMediaPackage().getIdentifier().toString();
+  private void updateWorkflowInstanceInIndex(WorkflowIndexData latestWorkflow, ElasticsearchIndex index) {
+    final long workflowInstanceId = latestWorkflow.getId();
+    final WorkflowState workflowState = WorkflowState.values()[latestWorkflow.getState()];
+    final String eventId = latestWorkflow.getMediaPackageId();
     final String orgId = securityService.getOrganization().getId();
     final User user = securityService.getUser();
 
     logger.debug("Updating workflow instance {} of event {} in the {} index.", workflowInstanceId, eventId,
             index.getIndexName());
     Function<Optional<Event>, Optional<Event>> updateFunction = (Optional<Event> eventOpt) -> {
       Event event = eventOpt.orElse(new Event(eventId, orgId));
-      event.setCreator(user.getName());
       event.setWorkflowId(workflowInstanceId);
-      event.setWorkflowDefinitionId(workflowInstance.getTemplate());
```

Comment:
Is it okay to not set these two?

Reply:
Well as far as we know it is. The future will show if anything unexpected goes wrong.